### PR TITLE
ticdc: add ticdc-avro-protocol.md

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/9449
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-avro-protocol.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+summary: Learn the concept of TiCDC Avro Protocol and how to use it.
+---
+
+# TiCDC Avro Protocol
+
+Avro is a data exchange format protocol defined by [Apache Avro™](https://avro.apache.org/) and chosen by [Confluent Platform](https://docs.confluent.io/platform/current/platform.html) as the default data exchange format. This document describes the implementation of the Avro data format in TiCDC, including TiDB extension fields, definition of the Avro data format, and the interaction between Avro and [Confluent Schema Registry](https://docs.confluent.io/platform/current/schema-registry/index.html).
+
+## Use Avro
+
+When using Message Queue (MQ) as downstream sink, you can specify Avro in `sink-uri`. TiCDC wraps Avro Messages with events as the basic unit and sends TiDB DML events downstream. When Avro detects a schema change, it registers the latest schema with Schema Registry.
```
Comment (zhangyangyu) [inline ticdc/ticdc-avro-protocol.md:12]: TiCDC wraps TiDB DML events into Avro messages and send them downstream. When Avro detects a schema change, it registers the latest schema into Schema Registry.

英文我觉得这样描述是最准确的，稍微和中文的行文有些差距。

Comment (zhangyangyu) [inline ticdc/ticdc-avro-protocol.md:12]: 1. TiCDC captures TiDB DML events and creates Avro messages from them, send these events downstream.
2. register the latest schema into schema registry.

File: ticdc/ticdc-avro-protocol.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+```shell
+cdc cli changefeed create --pd=http://127.0.0.1:2379 --changefeed-id="kafka-avro" --sink-uri="kafka://127.0.0.1:9092/topic-name?kafka-version=2.6.0&protocol=avro" --schema-registry=http://127.0.0.1:8081
+```
+
+The value of `--schema-registry` supports the https protocol and `username:password` authentication, for example, `--schema-registry=https://username:password@schema-registry-uri.com`. The username and password must be URL-encoded.
+
+## TiDB extension fields
+
+By default, Avro only includes data of the row where data changes in the DML event, excluding the type of data change or unique identifiers of TiDB-specific CommitTS  transactions. To address this issue, TiCDC introduces the following three TiDB extension fields to the Avro protocol. When `enable-tidb-extension` is set to `true` (defaults to `false`) in `sink-uri`, TiCDC generates Avro messages with these three fields:
```
Comment (zhangyangyu) [inline ticdc/ticdc-avro-protocol.md:26]: 1. data of the row where data changes -> data of changed rows
2. 我觉得 excluding 不太合适，应该是 with no
3. unique identifiers of TiDB-specific CommitTS transactions -> TiDB-specific CommitTS, the unique identifiers of transactions

File: ticdc/ticdc-avro-protocol.md (hunk 1 above)
Comment (shichun-0415) [reply ticdc/ticdc-avro-protocol.md:12]: > 1. TiCDC captures TiDB DML events and creates Avro messages from them, send these events downstream.
> 2. register the latest schema into schema registry.

1. accepted. Please synchronize the change to Chinese in https://github.com/pingcap/docs-cn/pull/9449.
2. "register with" is a fixed expression. See the picture captured from Confluent:
image

File: ticdc/ticdc-avro-protocol.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        "precision":10,
+        "scale":4,
+        "type":"bytes"
+    }
+}
+```
+
+## DDL events and schema changes
+
+Avro does not generate DDL events downstream. It checks whether a schema changes at each DML event. If a schema changes, Avro generates a new schema and registers it with the Schema Registry. If the schema change does not pass the compatibility check, the registration fails. Avro does not resolve any schema compatibility issues.
```
Comment (zhangyangyu) [inline ticdc/ticdc-avro-protocol.md:250]: register into

File: ticdc/ticdc-avro-protocol.md (hunk 1 above)
Comment (zhangyangyu) [reply ticdc/ticdc-avro-protocol.md:12]: Ohh, then it's fine.

File: ticdc/ticdc-avro-protocol.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+| ENUM       | ENUM      | string    |                                                                                                                           |
+| SET        | SET       | string    |                                                                                                                           |
+| DECIMAL    | DECIMAL   | bytes     | When `avro-decimal-handling-mode` is string, AVRO_TYPE is string.                                                         |
+
+In the Avro protocol, there are two other `sink-uri` parameters that affect the Column data format: `avro-decimal-handling-mode` and `avro-bigint-unsigned-handling-mode`.
+
+- `avro-decimal-handling-mode` controls how Avro handles decimal fields, including:
+
+    - string: Avro handles decimal fields as strings.
+    - precise：Avro stores decimal fields as bytes.
```
Comment (shichun-0415) [inline ticdc/ticdc-avro-protocol.md:184]:
```suggestion
    - precise: Avro handles decimal fields as bytes.
```

**Comment by zhangyangyu** [comment]
Nice catch! @qiancai

File: ticdc/ticdc-avro-protocol.md [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+The following is a configuration example:
+
+{{< copyable "shell-regular" >}}
+
+```shell
+cdc cli changefeed create --pd=http://127.0.0.1:2379 --changefeed-id="kafka-avro-enable-extension" --sink-uri="kafka://127.0.0.1:9092/topic-name?kafka-version=2.6.0&protocol=avro&avro-decimal-handling-mode=string&avro-bigint-unsigned-handling-mode=string" --schema-registry=http://127.0.0.1:8081
+```
+
+Most SQL Types are mapped to the base Column data format. Some other SQL types extend the base data format to provide more information.
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:199]:
```suggestion
Most SQL types are mapped to the base Column data format. Some other SQL types extend the base data format to provide more information.
```

File: ticdc/ticdc-avro-protocol.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+}
+```
+
+## DDL events and schema changes
+
+Avro does not generate DDL events downstream. It checks whether a schema changes each time a DML event occurs. If a schema changes, Avro generates a new schema and registers it with the Schema Registry. If the schema change does not pass the compatibility check, the registration fails. Avro does not resolve any schema compatibility issues.
+
+Note that, even if a schema change passes the compatibility check and a new version is registered, the data producers and consumers still need to perform an upgrade to ensure normal running of the system.
+
+Assume that the default compatibility policy of Confluent Schema Registry is `BACKWARD` and add a non-empty column to the source table. In this situation, Avro generates a new schema but fails to register it with Schema Registry due to compatibility issues. At this time, the changefeed enters error state.
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:254]:
```suggestion
Assume that the default compatibility policy of Confluent Schema Registry is `BACKWARD` and add a non-empty column to the source table. In this situation, Avro generates a new schema but fails to register it with Schema Registry due to compatibility issues. At this time, the changefeed enters an error state.
```

**Review by qiancai — APPROVED** [review]

**Review by zhangyangyu — APPROVED** [review]

**Pull request merged**